tests/test_elements/test_mol2mol/test_mol2scaffold.py

```diff
@@ -0,0 +1,104 @@
+"""Test the mol2scaffold module."""
+
+from typing import Any
+from unittest import TestCase
+
+from molpipeline import Pipeline
+from molpipeline.any2mol import AutoToMol
+from molpipeline.mol2any import MolToSmiles
+from molpipeline.mol2mol.scaffolds import MakeScaffoldGeneric, MurckoScaffold
+
+
+class TestMurckoScaffold(TestCase):
+    """Test the MurckoScaffold class."""
+
+    def test_murcko_scaffold_generation_pipeline(self) -> None:
+        """Test the scaffold generation."""
+        scaffold_pipeline = Pipeline(
+            steps=[
+                ("smiles_to_mol", AutoToMol()),
+                ("murcko_scaffold", MurckoScaffold()),
+                ("scaffold_to_smiles", MolToSmiles()),
+            ]
+        )
+        smiles_list = ["Cc1ccc(=O)[nH]c1", "O=CC1CCC(c2ccccc2)CC1", "CCC"]
+        expected_scaffold_list = ["O=c1cccc[nH]1", "c1ccc(C2CCCCC2)cc1", ""]
+
+        scaffold_list = scaffold_pipeline.transform(smiles_list)
+        for scaffold, expected_scaffold in zip(scaffold_list, expected_scaffold_list):
+            self.assertEqual(expected_scaffold, scaffold)
+
+
+class TestMakeScaffoldGeneric(TestCase):
+    """Test the MakeScaffoldGeneric class."""
+
+    def setUp(self) -> None:
+        """Set up the pipeline and common variables."""
+        self.generic_scaffold_pipeline = Pipeline(
+            steps=[
+                ("smiles_to_mol", AutoToMol()),
+                ("murcko_scaffold", MurckoScaffold()),
+                ("make_scaffold_generic", MakeScaffoldGeneric()),
+                ("scaffold_to_smiles", MolToSmiles()),
+            ]
+        )
+        self.smiles_list = ["Cc1ccc(=O)[nH]c1", "O=CC1CCC(c2ccccc2)CC1", "CCC"]
+
+    def check_generic_scaffold(
+        self, params: dict[str, Any], expected_scaffold_list: list[str]
+    ) -> None:
+        """Helper function to set parameters and check the results.
+
+        Parameters
+        ----------
+        params: dict[str, Any]
+            Parameters to set for the pipeline.
+        expected_scaffold_list: list[str]
+            Expected output of the pipeline.
+        """
+        self.generic_scaffold_pipeline.set_params(**params)
+        generic_scaffold_list = self.generic_scaffold_pipeline.transform(
+            self.smiles_list
+        )
+        for generic_scaffold, expected_generic_scaffold in zip(
```

Comment:
check for the same length of the iterables you give to `zip` like explained in my above comment

Reply:
switched to self.assertListEqual.